users.js

```diff
@@ -1231,41 +1300,47 @@ User = (function () {
 			this.updateIdentity(roomid);
 		}
 	};
-	User.prototype.ban = function (noRecurse, userid) {
+	User.prototype.ban = function (noRecurse, userid, time) {
 		// recurse only once; the root for-loop already bans everything with your IP
 		if (!userid) userid = this.userid;
+		if (!time) time = Date.now();
 		if (!noRecurse) {
 			for (var i in users) {
 				if (users[i] === this) continue;
 				for (var myIp in this.ips) {
 					if (myIp in users[i].ips) {
-						users[i].ban(true, userid);
+						users[i].ban(true, userid, time);
 						break;
 					}
 				}
 			}
 		}
 
+		var entry = {
+			time: time,
+			userid: userid
+		};
```

Comment:
Having entries being structured like this increases RAM usage and slows down access by a bit.

With 300ish active, it shouldn't be a big problem, though. We _do_ periodically clean them up, right?